src/Protocol/ActivityPub/Transmitter.php

```diff
@@ -0,0 +1,910 @@
+		if (!DBA::isResult($contact)) {
+			return [];
+		}
+
+		$data = ['@context' => ActivityPub::CONTEXT];
+		$data['id'] = $contact['url'];
+		$data['diaspora:guid'] = $user['guid'];
+		$data['type'] = ActivityPub::ACCOUNT_TYPES[$user['account-type']];
+		$data['following'] = System::baseUrl() . '/following/' . $user['nickname'];
+		$data['followers'] = System::baseUrl() . '/followers/' . $user['nickname'];
+		$data['inbox'] = System::baseUrl() . '/inbox/' . $user['nickname'];
+		$data['outbox'] = System::baseUrl() . '/outbox/' . $user['nickname'];
+		$data['preferredUsername'] = $user['nickname'];
+		$data['name'] = $contact['name'];
+		$data['vcard:hasAddress'] = ['@type' => 'vcard:Home', 'vcard:country-name' => $profile['country-name'],
+			'vcard:region' => $profile['region'], 'vcard:locality' => $profile['locality']];
+		$data['summary'] = $contact['about'];
+		$data['url'] = $contact['url'];
+		$data['manuallyApprovesFollowers'] = in_array($user['page-flags'], [Contact::PAGE_NORMAL, Contact::PAGE_PRVGROUP]);
+		$data['publicKey'] = ['id' => $contact['url'] . '#main-key',
+			'owner' => $contact['url'],
+			'publicKeyPem' => $user['pubkey']];
+		$data['endpoints'] = ['sharedInbox' => System::baseUrl() . '/inbox'];
+		$data['icon'] = ['type' => 'Image',
+			'url' => $contact['avatar']];
+
+		// tags: https://kitty.town/@inmysocks/100656097926961126.json
+		return $data;
+	}
+
+	/**
+	 * @brief Returns an array with permissions of a given item array
+	 *
+	 * @param array $item
+	 *
+	 * @return array with permissions
+	 */
+	private static function fetchPermissionBlockFromConversation($item)
+	{
+		if (empty($item['thr-parent'])) {
+			return [];
+		}
+
+		$condition = ['item-uri' => $item['thr-parent'], 'protocol' => Conversation::PARCEL_ACTIVITYPUB];
+		$conversation = DBA::selectFirst('conversation', ['source'], $condition);
+		if (!DBA::isResult($conversation)) {
+			return [];
+		}
+
+		$activity = json_decode($conversation['source'], true);
+
+		$actor = JsonLD::fetchElement($activity, 'actor', 'id');
+		$profile = APContact::getByURL($actor);
+
+		$item_profile = APContact::getByURL($item['author-link']);
+		$exclude[] = $item['author-link'];
+
+		if ($item['gravity'] == GRAVITY_PARENT) {
+			$exclude[] = $item['owner-link'];
+		}
+
+		$permissions['to'][] = $actor;
+
+		foreach (['to', 'cc', 'bto', 'bcc'] as $element) {
+			if (empty($activity[$element])) {
+				continue;
+			}
+			if (is_string($activity[$element])) {
+				$activity[$element] = [$activity[$element]];
+			}
+
+			foreach ($activity[$element] as $receiver) {
+				if ($receiver == $profile['followers'] && !empty($item_profile['followers'])) {
+					$receiver = $item_profile['followers'];
+				}
+				if (!in_array($receiver, $exclude)) {
+					$permissions[$element][] = $receiver;
+				}
+			}
+		}
+		return $permissions;
+	}
+
+	/**
+	 * @brief Creates an array of permissions from an item thread
+	 *
+	 * @param array $item
+	 *
+	 * @return permission array
```

Comment:
for consistency please use the return format @return type description - `@return array with permission`